x/ibc/04-channel/client/utils/utils.go

```diff
@@ -9,45 +9,57 @@ import (
 
 // QueryPacket returns a packet from the store
 func QueryPacket(
-	ctx client.CLIContext, portID, channelID string, sequence, timeout uint64, prove bool,
+	ctx client.CLIContext, portID, channelID string,
+	sequence, timeout uint64, queryRoute string,
 ) (types.PacketResponse, error) {
+	var packetRes types.PacketResponse
+
 	req := abci.RequestQuery{
 		Path:  "store/ibc/key",
 		Data:  types.KeyPacketCommitment(portID, channelID, sequence),
-		Prove: prove,
+		Prove: true,
 	}
 
 	res, err := ctx.QueryABCI(req)
 	if err != nil {
-		return types.PacketResponse{}, err
+		return packetRes, err
 	}
 
-	channelRes, err := QueryChannel(ctx, portID, channelID, prove)
+	channel, err := QueryChannel(ctx, portID, channelID, queryRoute, true)
 	if err != nil {
-		return types.PacketResponse{}, err
+		return packetRes, err
 	}
 
-	destPortID := channelRes.Channel.Counterparty.PortID
-	destChannelID := channelRes.Channel.Counterparty.ChannelID
+	destPortID := channel.Channel.Counterparty.PortID
+	destChannelID := channel.Channel.Counterparty.ChannelID
+
+	var data types.PacketDataI
+	// TODO: commitment data is stored, not the data
```

Comment:
Do you mean that a commitment is stored, not the full packet data? That should be the case. I don't really understand this comment - why are we storing the full data?

Reply:
The stored form is obtained by `hash(GetData(), GetTimeout())`, so we cannot unmarshal the full packet from it, the full packet should be obtainable from somewhere else(could be the state)

Reply:
Is this still an issue?

Reply:
> The stored form is obtained by `hash(GetData(), GetTimeout())`, so we cannot unmarshal the full packet from it, the full packet should be obtainable from somewhere else(could be the state)

Ah yes, I agree, I was just confused by the `PacketDataI` name, but it turns out that interface doesn't actually contain the data. Can we rename to `PacketCommitmentI` or something?

Reply:
> Ah yes, I agree, I was just confused by the `PacketDataI` name, but it turns out that interface doesn't actually contain the data. Can we rename to `PacketCommitmentI` or something?

Isn't the interface itself the data? Is this resolved with the new `GetBytes()` function you added @cwgoes ?

Reply:
It's resolved for now. I'm still not a huge fan of these interfaces, they need to be split at least - https://github.com/cosmos/cosmos-sdk/issues/5509